sdks/java/io/mongodb/src/test/java/org/apache/beam/sdk/io/mongodb/MongoDbIOTest.java

```diff
@@ -0,0 +1,167 @@
+
+  private static final int PORT = 27017;
+  private static final String DATABASE = "beam";
+  private static final String COLLECTION = "test";
+
+  private MongodExecutable mongodExecutable;
+
+  @Before
+  public void setup() throws Exception {
+    LOGGER.info("Starting MongoDB embedded instance");
+    IMongodConfig mongodConfig = new MongodConfigBuilder()
+        .version(PRODUCTION)
+        .net(new Net(PORT, Network.localhostIsIPv6()))
+        .build();
+    mongodExecutable = MongodStarter.getDefaultInstance().prepare(mongodConfig);
+    mongodExecutable.start();
+
+    LOGGER.info("Insert test data");
+
+    MongoClient client = new MongoClient("0.0.0.0", PORT);
+    MongoDatabase database = client.getDatabase(DATABASE);
+
+    MongoCollection collection = database.getCollection(COLLECTION);
+    collection.drop();
+
+    String[] scientists = {"Einstein", "Darwin", "Copernicus", "Pasteur", "Curie", "Faraday",
+        "Newton", "Bohr", "Galilei", "Maxwell"};
+    for (int i = 1; i <= 1000; i++) {
+      int index = i % scientists.length;
+      Document document = new Document();
+      document.append("_id", i);
+      document.append("scientist", scientists[index]);
+      collection.insertOne(document);
+    }
+
+  }
+
+  @After
+  public void stop() throws Exception {
+    LOGGER.info("Stopping MongoDB instance");
+    mongodExecutable.stop();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testFullRead() throws Exception {
+    TestPipeline pipeline = TestPipeline.create();
+
+    PCollection<String> output = pipeline.apply(
+        MongoDbIO.read()
+          .withUri("mongodb://localhost:" + PORT)
+          .withDatabase(DATABASE)
+          .withCollection(COLLECTION));
+
+    PAssert.thatSingleton(output.apply("Count", Count.<String>globally()))
+        .isEqualTo(new Long(1000));
+
+    pipeline.run();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testReadWithFilter() throws Exception {
+    TestPipeline pipeline = TestPipeline.create();
+
+    PCollection<String> output = pipeline.apply(
+        MongoDbIO.read()
+        .withUri("mongodb://localhost:" + PORT)
+        .withDatabase(DATABASE)
+        .withCollection(COLLECTION)
+        .withFilter("{\"scientist\":\"Einstein\"}"));
+
+    PAssert.thatSingleton(output.apply("Count", Count.<String>globally()))
+        .notEqualTo(new Long(1000));
+
+    pipeline.run();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testWrite() throws Exception {
+    TestPipeline pipeline = TestPipeline.create();
+
+    ArrayList<String> data = new ArrayList<>();
+    for (int i = 0; i < 100; i++) {
+      data.add("{\"scientist\":\"Test\"}");
+    }
+    pipeline.apply(Create.of(data))
+        .apply(MongoDbIO.write().withUri("mongodb://localhost:" + PORT).withDatabase("test")
+            .withCollection("test"));
+
+    pipeline.run();
+
+    MongoClient client = new MongoClient("localhost", PORT);
```

Comment:
As I would like to test `write` here, I prefer to use core MongoDB mechanism to validate the test. I think it's more secure and reliable (for instance if there's an issue in `MongoDbIO.read()`. WDYT ?